tests/ui/unnecessary_reserve.rs

```diff
@@ -0,0 +1,115 @@
+#![warn(clippy::unnecessary_reserve)]
+#![feature(custom_inner_attributes)]
+
+#[allow(clippy::unnecessary_operation)]
+use std::collections::{HashMap, VecDeque};
+
+fn main() {
+    vec_reserve();
+    vec_deque_reserve();
+    hash_map_reserve();
+    insufficient_msrv();
+    box_vec_reserve();
+}
+
+fn vec_reserve() {
+    let mut vec: Vec<usize> = vec![];
+    let array1: &[usize] = &[1, 2];
+    let array2: &[usize] = &[3, 4];
+
+    // do not lint - different arrays
+    vec.reserve(array1.len());
+    vec.extend(array2);
+
+    // do not lint
+    vec.reserve(1);
+    vec.extend([1]);
```

Comment:
Btw, error markers are now required in order for tests to pass. If you rebase over master, you will see examples in all the UI test files.